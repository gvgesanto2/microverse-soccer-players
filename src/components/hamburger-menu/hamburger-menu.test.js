import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import HamburgerMenu from './hamburger-menu.component';

const ACTIVE_HAMBURGER_MENU_CLASS = 'hamburger-menu--is-active';
const HAMBURGER_MENU_TEST_ID = 'hamburger-menu';

let component;
let hamburgerMenuHtmlElem;
let hamburgerButton;

describe('HamburgerMenu component', () => {
  beforeEach(() => {
    component = render(<HamburgerMenu />, { wrapper: BrowserRouter });
    hamburgerMenuHtmlElem = component.getByTestId(HAMBURGER_MENU_TEST_ID);
    hamburgerButton = screen.getByRole('button', { name: 'hamburger button' });
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should be closed by default', () => {
    const hamburgerMenuClassList = Object.values(hamburgerMenuHtmlElem.classList);
    expect(hamburgerMenuClassList).not.toContain(ACTIVE_HAMBURGER_MENU_CLASS);
  });

  it('should be able to toggle the menu clicking at the hamburger button', () => {
    userEvent.click(hamburgerButton);

    expect(Object.values(hamburgerMenuHtmlElem.classList)).toContain(ACTIVE_HAMBURGER_MENU_CLASS);

    userEvent.click(hamburgerButton);

    expect(
      Object.values(hamburgerMenuHtmlElem.classList),
    ).not.toContain(ACTIVE_HAMBURGER_MENU_CLASS);
  });

  it('should navigate to the homepage and close the menu when clicking the home navigation link', () => {
    const HomeNavLink = screen.getByRole('link', { name: /01 home/i });

    userEvent.click(hamburgerButton);
    userEvent.click(HomeNavLink);

    expect(
      Object.values(hamburgerMenuHtmlElem.classList),
    ).not.toContain(ACTIVE_HAMBURGER_MENU_CLASS);
    expect(window.location.pathname).toBe('/');
  });
});
