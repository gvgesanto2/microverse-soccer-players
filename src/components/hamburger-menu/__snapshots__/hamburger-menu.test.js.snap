// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hamburger-menu "
        data-testid="hamburger-menu"
      >
        <button
          aria-label="hamburger button"
          class="hamburger-btn "
          type="button"
        >
          <span
            class="hamburger-btn__icon"
          >
             
          </span>
        </button>
        <div
          class="hamburger-menu__background"
        >
           
        </div>
        <nav
          class="hamburger-menu__nav"
        >
          <ul
            class="hamburger-menu__nav-list"
          >
            <li>
              <a
                class="hamburger-menu__link"
                href="/"
              >
                <span>
                  01
                </span>
                home
              </a>
            </li>
            <li>
              <a
                class="hamburger-menu__link"
                href="https://github.com/gvgesanto2/microverse-soccer-players"
                rel="noopener noreferrer"
                target="__blank"
              >
                <span>
                  02
                </span>
                source code
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hamburger-menu "
      data-testid="hamburger-menu"
    >
      <button
        aria-label="hamburger button"
        class="hamburger-btn "
        type="button"
      >
        <span
          class="hamburger-btn__icon"
        >
           
        </span>
      </button>
      <div
        class="hamburger-menu__background"
      >
         
      </div>
      <nav
        class="hamburger-menu__nav"
      >
        <ul
          class="hamburger-menu__nav-list"
        >
          <li>
            <a
              class="hamburger-menu__link"
              href="/"
            >
              <span>
                01
              </span>
              home
            </a>
          </li>
          <li>
            <a
              class="hamburger-menu__link"
              href="https://github.com/gvgesanto2/microverse-soccer-players"
              rel="noopener noreferrer"
              target="__blank"
            >
              <span>
                02
              </span>
              source code
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
